source/rds.rst

```diff
@@ -0,0 +1,202 @@
+instance.)
+
+We also store each of these separate Postgres user's credentials in Lastpass
+in a shared folder associated with the corresponding client and
+project.
+
+Security and access to RDS servers
+----------------------------------
+
+We configure RDS servers to not be accessible from the Internet,
+but only from the EC2 instances hosting the servers that need to access
+them when running. That can be done by putting the EC2 servers in a
+common security group, and setting up the RDS server's security group to
+only allow port 5432 access from the EC2 servers' security group.
+
+This implies that developers and admins will not be able to do Postgres
+administration from their own machines, but will have to ssh to one of
+the permitted EC2 instances and do it from there.
+
+Provisioning databases for projects
+-----------------------------------
+
+When we're deploying a project that hosts its own Postgres server, often we'll
+create the database user and database for the project during the initial deploy.
+We will not be able to do that when the database is on RDS, as the deploy
+will not have the credentials of the master Postgres user.
+
+So, we create the database user and database manually (or using
+some separate bit of automation).  And that will need to be done from one of
+the EC2 instances. One approach would be to ssh into one of the servers and
+use the Postgres command line interface. Just be careful not to leave any
+credentials for the master user on that server. (If you want to be especially
+careful, use a temporary EC2 instance just for that, and destroy it when done.)
+
+See below for some detailed documentation on using the Postgres command
+line to do the things that will need to be done.
+
+Backups
+-------
+
+TBD!
+
+(Do we just rely on the automatic backups on RDS, which are only kept up
+to 45 days; do we make "manual" (initiated when we choose) backups on RDS,
+which will be kept as long as we want; or do we want to make backups
+outside of RDS?)
+
+Practical tips
+~~~~~~~~~~~~~~
+
+Before starting to work with RDS, read or at least skim through
+`the AWS documentation on how Postgres works on RDS
+<http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_PostgreSQL.html>`_,
+and pay special attention to
+`Appendix: Common DBA Tasks for PostgreSQL <http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Appendix.PostgreSQL.CommonDBATasks.html>`_.
+
+It'll save you a lot of headaches in the long run.
+
+Postgres on RDS
+---------------
+
+The master user that RDS gives you is *not* a Postgres superuser. Though it
+has a lot of permissions, there are a few things that you just won't be able
+to do quite the way you could if you were a superuser.
+
+In the examples below, for readability I'm omitting most of the common
+arguments to specify where the Postgres server is, what the database name is,
+etc. You can set some environment variables to use as defaults for things::
+
+    export PGDATABASE=dbname
```

Comment:
I like the convention of prefixing shell commands with `$` and `(envname) $` to make it clear they are run in the shell (with or without the virualenv) rather than code examples. I don't think we are always consistent about it but we should try to be.